#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.3107058566186369 0.0233473205155108 -0.9502193290433664 2.999021140050051
  position 2.230658393875791 0.3707306487444493 1.4399101666221217
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
  floorTileSize 0.1 0.1
}
DEF body Robot {
  translation 0.05020339104230199 0.5826452558425443 0.02183140141431402
  rotation -0.8965664406143723 0.44290911120479975 0.00036983219112149917 -3.140136730085061
  children [
    DistanceSensor {
      translation -0.02 -0.041 -0.017
      rotation 0 0 -1 2.2252956132800117
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightBack"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.02 -0.041 -0.017
      rotation 0 0 -1 0.9162996938995747
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftBack"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation -0.044 0.01 -0.017
      rotation 0 0 -1 3.14
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightSide"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.044 0.01 -0.017
      rotation 0 0 -1 0
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftSide"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation -0.02 0.041 -0.017
      rotation 0 0 -1 -1.9634996938995748
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSRightFront"
      numberOfRays 2
      aperture 0.6
    }
    DistanceSensor {
      translation 0.02 0.041 -0.017
      rotation 0 0 -1 -1.1765
      children [
        Solid {
          translation 0.004 0 0
          rotation 0 1 0 1.5708
          children [
            Shape {
              geometry Cone {
                bottomRadius 0.004
                height 0.01
              }
            }
          ]
        }
      ]
      name "DSLeftFront"
      numberOfRays 2
      aperture 0.6
    }
    Solid {
      translation 0 0 -0.016
      rotation 0 0 1 3.14159
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Cylinder {
            bottom FALSE
            height 0.014
            radius 0.051
            top FALSE
            subdivision 360
          }
        }
      ]
      name "head"
      boundingObject Cylinder {
        height 0.007
        radius 0.05
      }
      physics Physics {
      }
      rotationStep 0.3
    }
    DEF WL HingeJoint {
      jointParameters HingeJointParameters {
        position 147.07132130804357
        anchor 0.05 0 0.002
      }
      device [
        RotationalMotor {
          name "LWM"
        }
      ]
      endPoint Solid {
        translation 0.05 0 0.002
        rotation 0.6997905336456534 0.16108812108827764 0.6959481491204502 2.816608366088191
        children [
          DEF LW Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 0 0
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE LW
        physics Physics {
        }
        linearVelocity -0.07945688954733536 -0.06079987632528228 -8.618347423157788e-06
        angularVelocity 3.038513478001721 -3.9708389385216454 -0.000551198580189937
      }
    }
    DEF WR HingeJoint {
      jointParameters HingeJointParameters {
        position 148.9559106653189
        anchor -0.05 0 0.002
      }
      device [
        RotationalMotor {
          name "RWM"
        }
      ]
      endPoint Solid {
        translation -0.05 0 0.002
        rotation -0.6226705373872482 0.47387757426303884 -0.6226728245880445 2.256531217558885
        children [
          DEF RW Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 0 0
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE RW
        physics Physics {
        }
        linearVelocity -0.07941755031709997 -0.060770456991520616 7.012805487739618e-06
        angularVelocity 3.038513541718079 -3.9708389637066976 -0.0005512838908928253
      }
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.2 0.4 0.2
          emissiveColor 0 0.04 0.1
        }
      }
      geometry DEF body Cylinder {
        height 0.0435
        radius 0.05
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "Cylinder_Controller"
  linearVelocity -0.0794368512300618 -0.06078573329675786 -8.018372082299623e-07
  angularVelocity 0.00013403010686697901 8.664947780944177e-05 -0.0004563847414358965
}
